### ISSUE TITLE
Locked block toolbar layout broken when 'Show button text labels' is enabled

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#49865]
### Description

When the  'Show button text labels' preference is enabled, a locked block toolbar doesn't look right. 

There's no separation between the block switcher button and the unlock button. Visually, it's pretty confusing. Also the paddign on the left is incorrect.

### Step-by-step reproduction instructions

- Lock a paragraph block:
  - From the block toolbar > Options > Lock > Lock all > Apply 
- The block toolbar now shows the 'unlock' button after the block switcher:

<img width="468" alt="Screenshot 2023-04-17 at 15 47 43" src="https://user-images.githubusercontent.com/1682452/232505782-376512af-8a6c-4b1d-a667-b3ece03aebd5.png">

- Go to the editor Options > Preferences, and enable 'Show button text labels'.
- The block toolbar now shows text labels instead of icons.
- Observe there's no sepaaration between the first two buttons. Also, the left padding looks bigger than expected:

<img width="597" alt="Screenshot 2023-04-17 at 15 48 24" src="https://user-images.githubusercontent.com/1682452/232506221-e004c90c-5650-49a5-8164-939366f4dfab.png">

- It happens also with 'top toolbar' enabled:

<img width="588" alt="Screenshot 2023-04-17 at 15 51 47" src="https://user-images.githubusercontent.com/1682452/232506486-e21b44d7-d3f2-4ba2-9cc8-05996b050640.png">

For reference, here's how the block toolbar looks when the block is not locked:

<img width="690" alt="Screenshot 2023-04-17 at 15 49 37" src="https://user-images.githubusercontent.com/1682452/232506681-abbe7e22-a0a1-475e-b5c0-86046c195d1b.png">

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by annezazu** [comment]
@tellthemachines  curious if you could dive in here as I know you've done some fixes on the text label experience <3

**Comment by joedolson** [comment]
This is coming from this CSS:

`.block-editor-block-lock-toolbar .components-button.has-icon { padding-left: 0 !important; }`

It's removing the left padding on the lock button, which you can see has compressed spacing both when using icons and using text; but it makes a much bigger difference to comprehension when viewing as text.

Separately - is it necessary for that button to contain the block name? That context should already be apparent, since the neighboring button gives the context. Seems unnecessarily verbose.

**Comment by Mamaduka** [comment]
Created PR to resolve styling issues.

> Separately - is it necessary for that button to contain the block name? That context should already be apparent, since the neighboring button gives the context. Seems unnecessarily verbose.

I like the idea and am happy to create PR if everyone agrees on the change.

**Comment by afercia** [comment]
> Separately - is it necessary for that button to contain the block name? That context should already be apparent, since the neighboring button gives the context. Seems unnecessarily verbose.

I'd agree to try to remove the block name. However, I think that button needs a few more improvements so yes I'd recommend a separate PR.

Regardless whether it's icon or label, when a block is locked that button gets disabled with both a `disabled` attribute and `aria-disabled="true"`.

<img width="979" alt="Screenshot 2023-05-26 at 10 25 23" src="https://github.com/WordPress/gutenberg/assets/1682452/ae2f9ec3-3d74-4d69-825b-d394cb0a7618">

<img width="985" alt="Screenshot 2023-05-26 at 10 24 55" src="https://github.com/WordPress/gutenberg/assets/1682452/b86bafd1-d534-4d64-91c1-4735dc5e35b7">


That's not ideal because the button can't be navigated to with the keyboard. As such, there's no contextual information for screen reader users (although screen readers do provide alternative ways to get to that button). 

In several places in the Editor we do use a different pattern to keep disabled controls discoverable. See [Focusability of disabled controls](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#focusabilityofdisabledcontrols) in the ARIA practices. 

To recap:
- The block toolbar is an ARIA toolbar.
- Expected keyboard interaction: navigation through the toolbar buttons is by arrow keys.
- However, the 'Paragraph' button is not focusable because of the `disabled` attribute: the button gets skipped when arrowing.
- Screen reader users will miss that context.
- To remediate:
- The button should only use `aria-disabled="true"` and be 'noop-ed`. We should make sure that arrow navigation works on this button.
- When the block is locked, the button should also _look_ disabled. Currently, it doesn't.
- The icon button should still display its tooltip.
- The tooltip should not be displayed when 'Show button text labels' is enabled.

**Comment by Mamaduka** [comment]
I created a separate PR to make the button label less verbose - #51083.

@afercia, the block switcher button can be disabled for other reasons as well, so it's not tied to block locking. If you prefer, we can open a separate issue.

> When the block is locked, the button should also look disabled. Currently, it doesn't.

The button also serves as a visual clue for the block type; this is probably why disabled styles aren't applied.

**Comment by afercia** [comment]
Thanks @Mamaduka! I will create a separate issue for the disabled state.
